Kickstarter-iOS/Features/PledgePaymentMethods/Controller/NoShippingPledgeRewardsSummaryViewController.swift

```diff
@@ -184,6 +189,69 @@ final class NoShippingPledgeRewardsSummaryViewController: UIViewController {
     self.pledgeTotalViewController.view.isHidden = isHidden
     self.separatorView.isHidden = isHidden
   }
+
+  private func applySnapshotToDataSource(from data: [PostCampaignRewardsSummaryItem]) {
+    var snapshot = NSDiffableDataSourceSnapshot<PledgeRewardsSummarySection, PledgeRewardsSummaryRow>()
+
+    let headerItemData = data.compactMap { item -> PledgeExpandableHeaderRewardCellData? in
+      guard case let .header(data) = item else { return nil }
+      return data
+    }
+
+    let rewardData = data.compactMap { item -> PledgeExpandableHeaderRewardCellData? in
+      guard case let .reward(data) = item else { return nil }
+      return data
+    }
+
+    let baseReward = rewardData[0]
```

Comment:
Sanity checking, is it guaranteed that this array will always have at least one item in it?